request.go

```diff
@@ -40,6 +40,7 @@ type Request struct {
 	minRetryDelay time.Duration
 	maxRetryDelay time.Duration
 	sleepFn       func(d time.Duration) <-chan time.Time
+	hasExpected   bool
```

Comment:
The name is a bit confusing because it also sounds like we have some `expected` value. Maybe rename it to `expectCalled`?

Also this block of fields is for retry stuff, let's move new field to another block, dedicated for similar stuff (flags and info about what was called already):

```
	bodySetter string
	typeSetter string
	forceType  bool
```